### ISSUE TITLE
login fails, but active sessions are created on fb account

### ISSUE THREAD

**Issue by sabrinashie** [carpedm20/fbchat#446]
## Description of the problem
Sorry about posting another login issue, but I can't seem to login even though the username and password are definitely correct, especially since I can see all 5 active sessions documented when I take a look at my FB profile. 

I am not sure what's causing "home" to not be part of the url after getting /login/save-device/cancel/. Traceback is below:

## Code to reproduce
```
#!/usr/bin/env python3

from fbchat import Client
import getpass
import logging
logging.basicConfig(level=logging.DEBUG)
client = Client("<email>", getpass.getpass(prompt="Password: "))
```
## Traceback
```
Logging in <email>...
INFO:client:Logging in <email>...
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): m.facebook.com:443
DEBUG:urllib3.connectionpool:https://m.facebook.com:443 "GET / HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:https://m.facebook.com:443 "POST /login.php?login_attempt=1 HTTP/1.1" 302 0
DEBUG:urllib3.connectionpool:https://m.facebook.com:443 "GET /login/save-device/?login_source=login&refsrc=https%3A%2F%2Fwww.facebook.com%2F&_rdr HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:https://m.facebook.com:443 "GET /login/save-device/cancel/ HTTP/1.1" 302 0
DEBUG:urllib3.connectionpool:https://m.facebook.com:443 "GET /?_rdr HTTP/1.1" 200 None
Attempt #1 failed, retrying
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/fbchat/_client.py", line 254, in login
    state = State.login(email, password, user_agent=user_agent)
  File "/usr/local/lib/python3.7/site-packages/fbchat/_state.py", line 139, in login
    "(Failed on url: {})".format(r.url)
fbchat._exception.FBchatUserError: Login failed. Check email/password. (Failed on url: https://m.facebook.com/?_rdr)
ERROR:client:Attempt #1 failed, retrying

```
And onward, 4 more times.
## Environment information
- Python version 3.7
- `fbchat` version 1.7.2

**Comment by skrobchik** [comment]
I'm having the same issue. The password and email is correct, I quadruple checked ten times already. Whenever I try to login with fbchat it will throw FBchatUserError and I'll see a new logged in device in Facebook settings.

**Comment by sabrinashie** [comment]
Actually, I found that if I just comment out the if case in _state.py (line 134) that checks for "home" in the url, the client is able to work and I can send messages to others. I also tried testing what happens if I pass in the wrong password (with line 134 commented out) and was given this error:
```
Logging in <email>...
INFO:client:Logging in <email>...
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): m.facebook.com:443
DEBUG:urllib3.connectionpool:https://m.facebook.com:443 "GET / HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:https://m.facebook.com:443 "POST /login.php?login_attempt=1 HTTP/1.1" 302 0
DEBUG:urllib3.connectionpool:https://m.facebook.com:443 "GET /login/?email=<email>&li=LBQgXUCrgo0n19wJKUEp_dmO&e=1348092&refsrc=https%3A%2F%2Fwww.facebook.com%2F&_rdr HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): www.facebook.com:443
DEBUG:urllib3.connectionpool:https://www.facebook.com:443 "GET / HTTP/1.1" 200 None
Attempt #1 failed, retrying
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/fbchat/_client.py", line 254, in login
    state = State.login(email, password, user_agent=user_agent)
  File "/usr/local/lib/python3.7/site-packages/fbchat/_state.py", line 135, in login
    return cls.from_session(session=session)
  File "/usr/local/lib/python3.7/site-packages/fbchat/_state.py", line 169, in from_session
    fb_dtsg = FB_DTSG_REGEX.search(r.text).group(1)
AttributeError: 'NoneType' object has no attribute 'group'
ERROR:client:Attempt #1 failed, retrying
```
so maybe that if case is actually unnecessary?

**Comment by EvolvedAwesome** [comment]
Yup having the same issue here. Looks like the latest build was broken with a recent facebook update.

**Comment by apepenkov** [comment]
Any info when will it be fixed?

**Comment by coolkit-carl** [comment]
Also unable to log in

**Comment by apepenkov** [comment]
Coolkit-carl, in _state.py (line 134) add in if 'or True'
That will solve the problem

**Comment by coolkit-carl** [comment]
> 'or True'

Okay, I succeeded. Thank you

**Comment by madsmtm** [comment]
As @kapi2289 [noted](https://github.com/carpedm20/fbchat/pull/449#pullrequestreview-263215615), the suggested fix was a rather temporary solution. I've fixed this properly in efc8776e709c960d27598a8443d4d479425c6984, by simply just adding an extra check for whether the URL was `/`.

Thanks for the detailed report, and for your patience, I've been on holiday for the past few weeks 😉

**Comment by CutAs** [comment]
Sorry if I'm coming back to this again but I'm running the latest version of fbchat and I'm getting 'check username/password' error again/still. Did I miss something?
